packages/atomic/src/components/search-box-suggestions/suggestions-common.ts

```diff
@@ -32,12 +33,17 @@ export interface SearchBoxSuggestionsBindings extends Bindings {
 
 export const dispatchSearchBoxSuggestionsEvent = (
   event: SearchBoxSuggestionsEvent,
-  element: Element
+  element: HTMLElement
 ) => {
-  const canceled = element.dispatchEvent(
+  element.dispatchEvent(
     buildCustomEvent('atomic/searchBoxSuggestion/register', event)
   );
-  if (canceled) {
+
+  const hasParentSearchBox = elementHasAncestorTag(
```

Comment:
how comes you changed this here? from a quick test, it seems to work with the `canceled` pattern but not with this new one for the old components (because of initialisation order)

Reply:
Benjamin made this recently, it allows to traverse the Shadow DOM

Reply:
I made `elementHasAncestorTag` a while ago, I can see that Benjamin added a `closest` method that checks for shadow DOM too. I'll see if I can replace with that

Reply:
oh my bad I meant to use the closest method then 😂 

Seems that we can merge these methods?

Reply:
woops, my bad!

Reply:
I actually didn't replace the other place that's using my method yet, because I'm not sure if traversing shadow doms will keep the behaviour that I want! so don't worry about it, I'll look at it